### ISSUE TITLE
Unable to Build Plugin

### ISSUE THREAD

**Issue by wolfenxx** [levnikmyskin/hyprland-virtual-desktops#58]
Hey guys

I have a flake.nix set up following your example almost to the letter in here (minor difference being the module is in a separate file) https://github.com/levnikmyskin/hyprland-virtual-desktops?tab=readme-ov-file#installing-on-nixos-with-homemanager, however I'm getting some build errors and I'm not sure how to resolve them
![20240818_17h38m16s_grim](https://github.com/user-attachments/assets/ede0b852-c0b1-4305-93aa-9273fd9d6d7c)

flake.nix
[flake.txt](https://github.com/user-attachments/files/16652875/flake.txt)

home.nix 
[home.txt](https://github.com/user-attachments/files/16652868/home.txt)

Any help would be appreciated

**Comment by levnikmyskin** [comment]
mh these are indeed weird errors. Are you on "rolling" nix os or stable nix os? Also what version of hyprland are you building?

@wiillou as usual, I'll ask for your help regarding nixos stuff :pray:

**Comment by wolfenxx** [comment]
@levnikmyskin it's unstable nixos
`   nixpkgs.url = "nixpkgs/nixos-unstable";
    home-manager.url = "github:nix-community/home-manager";
    home-manager.inputs.nixpkgs.follows = "nixpkgs";
    hyprland.url = "git+https://github.com/hyprwm/Hyprland?submodules=1";
    hyprland.follows = "hyprland-virtual-desktops/hyprland";
    hyprland-virtual-desktops.url = "github:levnikmyskin/hyprland-virtual-desktops";
`
I've tried rolling with stable release but same issue. I have a feeling there may be "crap" left over in my system from all this experimenting that's causing issues. I wanna confirm in a vm before I nuke my PC but hyprland on a vm is a bit weird

**Comment by barthofu** [comment]
Hey! Same problem here, i tested with the last version 2.2.4 (which btw shows as 2.2.2 as it has't been updated in the `flake.nix` file.

**Comment by barthofu** [comment]
Said nothing, i fixed it by adding `inputs.hyprland.follows = "hyprland"` in my flake input;
```nix
hyprland-virtual-desktops = {
      url = "https://github.com/levnikmyskin/hyprland-virtual-desktops/archive/refs/tags/v2.2.4.tar.gz";
      inputs.hyprland.follows = "hyprland";
};
```

**Comment by barthofu** [comment]
Thanks a lot for this plugin btw, it is a life saver!

**Comment by levnikmyskin** [comment]
yes `flake.nix` is not really maintained as I'm on arch and not on nixos. Please open a PR if this needs fixing :pray:

**Comment by wolfenxx** [comment]
@barthofu can you share the rest of your flake? I wanna see if I can repro it on my side (make it work I mean. I still fails for me)

**Comment by barthofu** [comment]
@wolfenxx yes np, it is accessible here -> https://github.com/barthofu/dotfiles

**Comment by levnikmyskin** [comment]
hey everyone following this issue:
I've just updated the flake so that it now pins hyprland to a specific commit, i.e., the latest tag release (I'll take care to update this every time).
I think this issue can be closed. Feel free to comment if something's wrong
